ui/src/main/resources/edu/wpi/grip/ui/templates/java/BasicStep.vm

```diff
@@ -0,0 +1,28 @@
+		//input
+#foreach($input in $step.getInputs())
+#if( $input.mutable() )##Is Mutable
+#if( $input.hasValue() )##Has a set value
+		$input.baseType() $input.name()$c = $input.value();
+#else##Has Connected Socket
+		$input.baseType() $input.name()$c = ${input.value()}.get().$input.baseType()Value();
+#end##end Has a set value
+#else##Not Mutable
+#if ( $input.type() == "FlipCode" )
+		$input.type() $input.name()$c = FlipCode.$input.value();
+#else
+		$input.type() $input.name()$c = $input.value();
+#end
+#end##end Is Mutable
+#end
```

Comment:
There is a bunch of special case logic here. It would be nice if this could be boiled down to something like the following by pushing the special cases into macros:

```
#foreach($input in $step.getInputs())
		#TypeDeclaration( $input ) #Identifier( $input, $c ) = #ValueExpression( $input )
#end
```